x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/action/InferModelAction.java

```diff
@@ -0,0 +1,188 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+package org.elasticsearch.xpack.core.ml.action;
+
+import org.elasticsearch.action.ActionRequest;
+import org.elasticsearch.action.ActionRequestBuilder;
+import org.elasticsearch.action.ActionRequestValidationException;
+import org.elasticsearch.action.ActionResponse;
+import org.elasticsearch.action.ActionType;
+import org.elasticsearch.client.ElasticsearchClient;
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.xpack.core.ml.inference.results.ClassificationInferenceResults;
+import org.elasticsearch.xpack.core.ml.inference.results.InferenceResults;
+import org.elasticsearch.xpack.core.ml.inference.results.RegressionInferenceResults;
+import org.elasticsearch.xpack.core.ml.utils.ExceptionsHelper;
+
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+
+public class InferModelAction extends ActionType<InferModelAction.Response> {
+
+    public static final InferModelAction INSTANCE = new InferModelAction();
+    public static final String NAME = "cluster:admin/xpack/ml/infer";
+
+    private InferModelAction() {
+        super(NAME, Response::new);
+    }
+
+    public static class Request extends ActionRequest {
+
+        private final String modelId;
+        private final long modelVersion;
+        private final List<Map<String, Object>> objectsToInfer;
+        private final boolean cacheModel;
+        private final Integer topClasses;
+
+        public Request(String modelId, long modelVersion) {
+            this(modelId, modelVersion, Collections.emptyList(), null);
+        }
+
+        public Request(String modelId, long modelVersion, List<Map<String, Object>> objectsToInfer, Integer topClasses) {
+            this.modelId = modelId;
+            this.modelVersion = modelVersion;
+            this.objectsToInfer = objectsToInfer == null ?
+                Collections.emptyList() :
+                Collections.unmodifiableList(objectsToInfer);
+            this.cacheModel = true;
```

Comment:
This seems like it is hardcoded to `true` at the moment. Just double-checking we still need it as part of the request.